corsheaders/checks.py

```diff
@@ -0,0 +1,97 @@
+import collections
+import numbers
+
+from django.core import checks
+from .conf import corsheaders_settings
+
+@checks.register
+def check_settings(app_configs, **kwargs):
+
+    errors = []
+    if not isinstance(corsheaders_settings.CORS_ALLOW_HEADERS, collections.Sequence):
+        errors.append(
+            checks.Error(
+                "CORS_ALLOW_HEADERS must be sequence",
+                id="corsheaders.W001"
+            )
+        )
+    if not isinstance(corsheaders_settings.CORS_ALLOW_METHODS, collections.Sequence):
+        errors.append(
+            checks.Error(
+                "CORS_ALLOW_METHODS must be sequence",
+                id="corsheaders.W002"
+            )
+        )
+    if not isinstance(corsheaders_settings.CORS_ALLOW_CREDENTIALS, bool):
+        errors.append(
+            checks.Error(
+                "CORS_ALLOW_CREDENTIALS must be bool",
+                id="corsheaders.W003"
+            )
+        )
+
+    if not isinstance(corsheaders_settings.CORS_PREFLIGHT_MAX_AGE, numbers.Integral):
+        errors.append(
+            checks.Error(
+                "CORS_PREFLIGHT_MAX_AGE must be integer",
+                id = "corsheaders.W004"
+            )
+        )
+    elif (corsheaders_settings.CORS_PREFLIGHT_MAX_AGE < 0):
+        errors.append(
+            checks.Error(
+                "CORS_PREFLIGHT_MAX_AGE cannot be negative number",
+                id = "corsheaders.W004.1"
```

Comment:
please only use the format `corsheaders.EDDD` where D is a digit.

You can combine these two into the same error anyway like

```python
if not isinstance(..) or corsheaders_settings... < 0:
    errors.append(... "CORS_PREFLIGHT_MAX_AGE should be an integer >= 0"
```